docs/charmsindetail.md

```diff
@@ -0,0 +1,304 @@
+# Charms in detail
+## Table of contents
+
+[Best practices](##Best%20practice)
+
+[Charm writing in detail](##Charm%20Writing%20in%20detail)
+
+[New to charms?](##New%20to%20charms?)
+
+[Charm Writing](#Charm%20writing)
+
+[Example Charms](##Example%20Charms)
+
+## Best practices
+
+### Repoository Naming
+
+Naming follows the convention of charm-<charm_name> for example `charm-percona-cluster`.
+
+The rationale behind this is quick and easy searching of the 1000s of github projects along with other reasons.
+
+
+## Charm Writing in detail
+
+## New to charms?
+
+> This covers aspects of charming which will be useful when creating them
```

Comment:
Most of this section should be in another file. It's useful background information, but isn't about charm writing.